### PR TITLE
WSTEAMA-308: Refactor how data is fetched from the BFF

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves JIRA https://jira.dev.bbc.co.uk/browse/WSTEAMA-308

Overall changes
======
Refactors all page types which fetch data from the BFF to use common, shared code - reducing duplication and improvig future maintainability

Code changes
======
- Addition of fetchDataFromBFF helper function
- Update all BFF page types (except for the live page - we ran into issues with integration tests so decided to leave it for now)
-  We now have a single place for all PAGE_TYPES, which works with both Simorgh & BFF

Testing
======
- [ ] All automated tests passing - unit / integration / cypress
- [ ] Visit at least 1 example of each page type in local environment - page renders as expected. Check logs to ensure that fixture data is fetched locally, but when renderer override is set, the data will be fetched from the BFF

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

**Review by amoore108 — APPROVED** [review]
Looks great! Thanks for doing this 🙏

File: ws-nextjs-app/pages/[service]/new_live/[id]/[[...variant]].page.tsx [hunk 1]
```diff
@@ -48,10 +48,8 @@ const getPageData = async ({
 
   const env = getEnvironment(pathname);
   const optHeaders = { 'ctx-service-env': env };
-  const isLocal = !env || env === 'local';
-  const certsNeeded = !isLocal && process.env.INTEGRATION_TEST_BUILD !== 'true';
 
-  const agent = certsNeeded ? await getAgent() : null;
+  const agent = certsRequired(pathname) ? await getAgent() : null;
```
Comment (HarveyPeachey) [inline ws-nextjs-app/pages/[service]/new_live/[id]/[[...variant]].page.tsx:52]: Probably worth us, (Content team) to have a ticket to get the reusable code to work with the live page

**Review by HarveyPeachey — APPROVED** [review]
LGTM!

File: ws-nextjs-app/pages/[service]/new_live/[id]/[[...variant]].page.tsx (hunk 1 above)
Comment (karinathomasbbc) [reply ws-nextjs-app/pages/[service]/new_live/[id]/[[...variant]].page.tsx:52]: Yeah, Aaron and I had a bit of a chat about this on Slack. I couldn't get the integration tests working, despite it working locally 😭. We also noticed that the integration tests for the live page only check for the heading & nav bar, so even if the page returned a 404, the integration tests would still pass. If you do pick up the ticket, you should be able to cherry-pick the commits from this PR (or maybe you might want to just start again! `git cherry-pick cf8c43f25574444a302bccb1e98fd66cc7249b27`)

**Pull request merged**